cmd/main.go

```diff
@@ -0,0 +1,155 @@
+package main
+
+import (
+	"context"
+	"log"
+	"net"
+	"net/http"
+	"path/filepath"
+	"strings"
+
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	_ "github.com/lib/pq"
+	"github.com/spf13/viper"
+	"golang.org/x/time/rate"
+	"google.golang.org/grpc"
+
+	"gwi_api/internal/app"
+	"gwi_api/internal/config"
+	"gwi_api/internal/repository"
+	"gwi_api/internal/service"
+	desc "gwi_api/pkg"
+	appruntime "runtime"
+)
+
+func main() {
+	// Gwi DB
+	gwidb, err := repository.NewDB()
+	if err != nil {
+		log.Fatalf("[ERR] cannot create database %s", err)
+		return
+	}
+
+	// preparing config file
+	configPath, err := config.GetConfigPath()
+	if err != nil {
+		log.Fatalln("cannot determine config path")
+	}
+	viper.AddConfigPath(configPath)
+	viper.SetConfigName("config")
+	err = viper.ReadInConfig()
+	if err != nil {
+		log.Fatalln("cannot read from a config")
+	}
+
+	//stripeKey := viper.Get("stripe.secret.key").(string)
+
+	// JWT
+	signedKeyJWT := viper.Get("jwt.signedKey").(string)
+	tokenManager := service.NewTokenManager(signedKeyJWT)
+
+	// Register all services
+	//dbHandler := repository.NewDBHandler(db)
+	gwiDB := repository.NewDBHandler(gwidb)
+	//logDBHandler := repository.NewDBHandler(logdb)
+
+	userService := service.NewUserService(gwiDB)
```

Comment:
Why do you pass all the available handlers (e.g. favourites, asserts) in the user's service? Could you do something better here?